src/main/java/seedu/address/storage/JsonAdaptedEvent.java

```diff
@@ -0,0 +1,162 @@
+package seedu.address.storage;
+
+import static java.time.temporal.TemporalAdjusters.next;
+import static seedu.address.model.recurfrequency.RecurFrequency.INVALID_RECUR_FREQUENCY_MESSAGE;
+
+import java.time.DateTimeException;
+import java.time.DayOfWeek;
+import java.time.Duration;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.TemporalAdjusters;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.commons.exceptions.InvalidEnumArgumentException;
+import seedu.address.model.recurfrequency.RecurFrequency;
+import seedu.address.model.schedule.Event;
+import seedu.address.model.schedule.EventDescription;
+
+/**
+ * Jackson-friendly version of {@link Event}.
+ */
+class JsonAdaptedEvent {
+
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Person's %s field is missing!";
+    public static final String MISSING_RECUR_FREQUENCY_CASE = "%s switch case is missing in JsonAdaptedEvent!";
+
+    private final String eventDescription;
+    private final String date;
+    private final String time;
+    private final String duration;
+    private final String recurFrequency;
+
+    /**
+     * Constructs a {@code JsonAdaptedEvent} with the given person details.
+     */
+    @JsonCreator
+    public JsonAdaptedEvent(@JsonProperty("eventDescription") String eventDescription,
+                             @JsonProperty("date") String date,
```

Comment:
Is this indentation one away?

Reply:
I think my identation might be off. I will fix this.